src/ansys/fluent/core/services/field_data.py

```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
     return fields_data
 
 
+class BaseFieldData:
+    """Contain common properties required by all field data types."""
+
+    def __init__(self, i_d, data):
+        self._data = data
+        self._id = i_d
+
+    @property
+    def data(self):
+        return self._data
+
+    @property
+    def surface_id(self):
+        return self._id
+
+    @property
+    def size(self):
+        return len(self._data)
+
+    def __getitem__(self, item):
+        return self._data[item]
+
+
+class ScalarFieldData(BaseFieldData):
+    """Container for scalar field data."""
+
+    class ScalarData:
+        """Store and provide the data as a scalar."""
```

Comment:
```suggestion
        """Stores and provides the data as scalar data."""
```
Because "scalar" is a noun, I added the noun "data" after it. However, this could possible be just "Stores and provides scalar data." Your call!